examples/http/azure_vm_list_metrics.js

```diff
@@ -402,270 +358,179 @@ function login() {
+ * @returns {Promise} A promise that resolves when the configuration for all VMs has been retrieved
  */
 function retrieveVMsConfiguration(vmInfoList) {
     const promises = vmInfoList.map(function (vmInfo) {
-        const d = D.q.defer();
-        const config = generateConfig("/resourceGroups/" + vmInfo.resourceGroup + "/providers/Microsoft.Compute/virtualMachines/" + vmInfo.vmName + "/instanceView?api-version=2024-07-01");
-        azureCloudManagementService.http.get(config, processVmConfigResponse(d, vmInfo));
-        return d.promise;
+        const d = D.q.defer()
+        const config = generateConfig("/resourceGroups/" + vmInfo.resourceGroup + "/providers/Microsoft.Compute/virtualMachines/" + vmInfo.vmName + "/instanceView?api-version=2024-07-01")
+        azureCloudManagementService.http.get(config, processVmConfigResponse(d, vmInfo))
+        return d.promise
     })
-    return D.q.all(promises);
+    return D.q.all(promises)
 }
 
 /**
- * Retrieves performance metrics for a list of virtual machines.
- * @param {Array} vmInfoList - The list of virtual machine information objects.
- * @returns {Promise} - A promise that resolves with an array of virtual machine information objects.
+ * Retrieves performance metrics for Azure VMs, grouped by performance metrics
+ * @param {Array} vmInfoList List of VM information objects to retrieve performance metrics for
+ * @returns {Promise} A promise that resolves once all performance data has been retrieved
  */
 function retrieveVMsPerformanceMetrics(vmInfoList) {
-    const performanceKeyGroups = [];
-    const maxGroupSize = 20;
-
+    const performanceKeyGroups = []
+    const promises = []
+    const maxGroupSize = 20
     for (let i = 0; i < performanceMetrics.length; i += maxGroupSize) {
         performanceKeyGroups.push(performanceMetrics.slice(i, i + maxGroupSize).map(function (metric) {
             return metric.key
-        }).join(','));
+        }).join(','))
     }
-
-    const promises = vmInfoList.map(function (vmInfo) {
-        const d = D.q.defer();
-
+    vmInfoList.forEach(function (vmInfo) {
+        const d = D.q.defer()
         if (vmInfo.displayStatus === "VM running") {
             const groupPromises = performanceKeyGroups.map(function (group) {
-                return new Promise(function (resolve, reject) {
-                    const config = generateConfig("/resourceGroups/" + vmInfo.resourceGroup + "/providers/Microsoft.Compute/virtualMachines/" + vmInfo.vmName + "/providers/microsoft.insights/metrics?api-version=2024-02-01&metricnames=" + group + "&timespan=PT1M");
-                    azureCloudManagementService.http.get(config, function (error, response, body) {
-                        processVmPerformanceResponse({
-                            resolve: resolve, reject: reject
-                        }, vmInfo)(error, response, body);
-                    });
-                });
-            });
+                const groupD = D.q.defer()
```

Comment:
we don't need subgroup of promises, use the promises variable, append to it all the promises created in this loop and remove the promise in the line 469

for this change use forEach instead of map